core-library/src/main/java/org/silverpeas/core/admin/user/dao/GroupSearchCriteriaForDAO.java

```diff
@@ -140,134 +127,117 @@ public GroupSearchCriteriaForDAO onMixedDomainOronDomainId(String domainId) {
   public SearchCriteria onUserIds(String... userIds) {
     if (userIds != ANY) {
       tables.add("st_group_user_rel");
-      StringBuilder[] sqlLists = asSQLList(userIds);
-      getFixedQuery().append("st_group.id = st_group_user_rel.groupid");
-      StringBuilder theQuery =
-          getFixedQuery().append(" and (st_group_user_rel.userid in ").append(sqlLists[0]);
-      for (int i = 1; i < sqlLists.length; i++) {
-        theQuery.append(" or st_group_user_rel.userid in ").append(sqlLists[i]);
-      }
-      theQuery.append(")");
+      this.userIds = Arrays.copyOf(userIds, userIds.length);
     }
     return this;
   }
 
   @Override
   public SearchCriteria onRoleNames(String... roleIds) {
     if (roleIds != null && roleIds.length > 0) {
-      tables.add("st_group");
       tables.add("st_userrole_group_rel");
-      StringBuilder[] sqlLists = asSQLList(roleIds);
-      StringBuilder theQuery = getFixedQuery().
-          append(
-              "(ST_Group.id = ST_UserRole_Group_Rel.groupId and (ST_UserRole_Group_Rel.userRoleId" +
-                  " in ").
-          append(sqlLists[0]);
-      for (int i = 1; i < sqlLists.length; i++) {
-        theQuery.append(" or ST_UserRole_Group_Rel.userRoleId in ").append(sqlLists[i]);
-      }
-      theQuery.append("))");
+      this.criteria.onRoleNames(roleIds);
     }
     return this;
   }
 
   public SearchCriteria onSuperGroupId(String superGroupId) {
     if (isDefined(superGroupId)) {
-      tables.add("st_group");
-      getFixedQuery().append("st_group.superGroupId = ").append(Integer.valueOf(superGroupId));
+      this.criteria.onSuperGroupId(superGroupId);
     }
     return this;
   }
 
   public SearchCriteria onAsRootGroup() {
-    tables.add("st_group");
-    getFixedQuery().append("st_group.superGroupId is null");
+    this.criteria.onAsRootGroup();
     return this;
   }
 
-  public String toSQLQuery(String fields) {
-    return MessageFormat.format(QUERY, fields, impliedTables(), queryFilter());
+  public JdbcSqlQuery toSQLQuery(String fields) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "toSQLQuery" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)